### ISSUE TITLE
FB7419552: Buttons having borderless- and link- styles became invisible in macOS 10.15.1

### ISSUE THREAD

**Issue by shpakovski** [feedback-assistant/reports#62]
- **Date:** 2019-10-31
- **Resolution:** Open
- **Area:** SwiftUI Framework
- **OS:** macOS 10.15.1
- **Type:** Incorrect/Unexpected Behavior
- **Keywords:** swiftui, buttons, links, appkit, regression

### Please describe the issue and what steps we can take to reproduce it:

Buttons having `LinkButtonStyle()` or `BorderlessButtonStyle()` are invisible:

1. Download a sample project, extract it, open in Xcode and launch
2. Your screen should look like `invisible-borderless-and-link-style-buttons.png`

## Files

[invisible-borderless-and-link-style-buttons.zip](https://github.com/feedback-assistant/reports/files/3792440/invisible-borderless-and-link-style-buttons.zip)

<img width="328" alt="invisible-borderless-and-link-style-buttons" src="https://user-images.githubusercontent.com/163410/67924334-7283a200-fbc1-11e9-9d1a-c24740c8eba5.png">

**Comment by shpakovski** [comment]
```swift
/// Replacement for AppKit while macOS 10.15.2 is released.
struct LinkButtonStyle: ButtonStyle {
  func makeBody(configuration: ButtonStyleConfiguration) -> some View {
    configuration.label
      .foregroundColor(.init(NSColor.linkColor.withSystemEffect(configuration.isPressed ? .pressed : .none)))
  }
}

/// Replacement for AppKit while macOS 10.15.2 is released.
struct BorderlessButtonStyle: ButtonStyle {
  func makeBody(configuration: ButtonStyleConfiguration) -> some View {
    configuration.label
      .foregroundColor(.init(NSColor.controlTextColor.withSystemEffect(configuration.isPressed ? .pressed : .none)))
  }
}
```

**Comment by shpakovski** [comment]
Confirmed fixed in macOS 10.15.2.
